src/common/cockpitcloserange.c

```diff
@@ -0,0 +1,85 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2021 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "config.h"
+#include "cockpitcloserange.h"
+
+#include <assert.h>
+#include <dirent.h>
+#include <err.h>
+#include <errno.h>
+#include <limits.h>
+#include <stddef.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/resource.h>
+
+int
+cockpit_close_range (int from, int max_fd, int flags)
+{
+  /* we want to keep the API compatible to glibc's upcoming close_range(), but don't implement flags or upper bounds */
+  assert (flags == 0);
+  assert (max_fd == INT_MAX);
+
+  DIR *d;
+
+  if ((d = opendir ("/proc/self/fd"))) {
+      struct dirent *de;
+
+      while ((de = readdir (d))) {
+          long l;
+          char *e = NULL;
+
+          errno = 0;
+          l = strtol (de->d_name, &e, 10);
```

Comment:
```suggestion
          long l = strtol (de->d_name, &e, 10);
```

Reply:
also: pretty sure you don't need to initialize `e`.

Reply:
Done both.